include/fiction/algorithms/simulation/sidb/random_layout_generator.hpp

```diff
@@ -0,0 +1,200 @@
+//
+// Created by Jan Drewniok on 05.04.23.
+//
+
+#ifndef FICTION_RANDOM_LAYOUT_GENERATOR_HPP
+#define FICTION_RANDOM_LAYOUT_GENERATOR_HPP
+
+#include "fiction/algorithms/path_finding/distance.hpp"
+#include "fiction/io/write_sqd_layout.hpp"
+#include "fiction/layouts/cell_level_layout.hpp"
+#include "fiction/technology/cell_technologies.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/technology/sidb_nm_position.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+
+#include <iostream>
+#include <vector>
+
+namespace fiction
+{
+/**
+ * This struct stores the parameters for the *generate_random_layouts* algorithm.
+ */
+template <typename Lyt>
+struct random_layout_params
+{
+    /**
+     * Aspect ratio of the layout (i.e. the size of the layout).
+     */
+    typename Lyt::aspect_ratio max_coordinate;
+    /**
+     * Number of SiDBs that are placed on the layout.
+     */
+    uint64_t number_placed_sidbs = 0;
+    /**
+     * If positively charged SiDBs should be prevented, SiDBs are not placed closer than the minimal_spacing.
+     */
+    bool prevent_positive_charges = true;
+    /**
+     * If positively charged SiDBs should be prevented, SiDBs are not placed closer than this value (2 cells as
+     * Euclidean distance by default).
+     */
+    double minimal_spacing = 2;
+    /**
+     * Maximal number of attempts to generate the layout.
+     */
+    uint64_t maximal_attempts = 10E6;
+};
+
+namespace detail
+{
+template <typename Lyt>
+class generate_random_layout_impl
+{
+  public:
+    generate_random_layout_impl(const random_layout_params<Lyt>& params, std::ostream& s,
+                                std::vector<Lyt>& all_layouts) :
+
+            parameter{params},
+            os{s},
+            previous_layouts{all_layouts}
+    {}
+
+    void run()
+    {
+        bool successful_generation = false;
+        while (!successful_generation && parameter.maximal_attempts)
+        {
+            // layout is initialized with given aspect ratio and name.
+            Lyt lyt{parameter.max_coordinate};
+            // maximal x-coordinate is stored.
+            auto x_dimension = parameter.max_coordinate.x;
+            // maximal y-coordinate is stored.
+            auto                   y_dimension = parameter.max_coordinate.y;
```

Comment:
These are just `lyt.x()` and `lyt.y()`, no?

Reply:
yes, it is a redundant step. Thanks!